jdk-httpserver/test/io/pedestal/http/jdk_httpserver_test.clj

```diff
@@ -0,0 +1,322 @@
+(ns io.pedestal.http.jdk-httpserver-test
+  (:refer-clojure :exclude [send])
+  (:require [cheshire.core :as json]
+            [clojure.core.async :as async]
+            [clojure.edn :as edn]
+            [clojure.java.io :as io]
+            [clojure.string :as string]
+            [clojure.test :refer [deftest is]]
+            [io.pedestal.connector :as conn]
+            [io.pedestal.http.jdk-httpserver]
+            [io.pedestal.http.route.definition.table :as table]
+            [io.pedestal.interceptor :as interceptor]
+            [io.pedestal.service.protocols :as p])
+  (:import (com.sun.net.httpserver HttpsConfigurator)
+           (java.lang AutoCloseable)
+           (java.net URI)
+           (java.net.http HttpClient HttpClient$Version HttpHeaders HttpRequest HttpRequest$BodyPublishers HttpResponse
+                          HttpResponse$BodyHandlers)
+           (java.nio ByteBuffer)
+           (java.nio.channels Pipe)
+           (java.security KeyStore)
+           (java.security.cert X509Certificate)
+           (java.time Duration)
+           (java.util Optional)
+           (java.util.function BiPredicate Supplier)
+           (javax.net.ssl KeyManagerFactory SSLContext TrustManager X509ExtendedTrustManager)))
+
+(set! *warn-on-reflection* true)
+
+(comment
+  (System/setProperty "io.pedestal.http.jdk-httpserver-test.create-connector"
+    (str `io.pedestal.http.jdk-httpserver/create-connector))
+
+  (System/setProperty "io.pedestal.http.jdk-httpserver-test.create-connector"
+    (str `io.pedestal.http.jetty/create-connector))
+
+  (System/setProperty "io.pedestal.http.jdk-httpserver-test.create-connector"
+    (str `io.pedestal.http.http-kit/create-connector)))
+
+(defn create-and-start!
+  ^AutoCloseable [connector-map connector-opts]
+  (let [create-connector (-> "io.pedestal.http.jdk-httpserver-test.create-connector"
+                           (System/getProperty "io.pedestal.http.jdk-httpserver/create-connector")
+                           symbol
+                           requiring-resolve
+                           (or (throw (ex-info "Can't find create-connector" {}))))
+        conn (-> connector-map
+               (create-connector connector-opts)
+               conn/start!)]
+    (vary-meta (reify
+                 AutoCloseable
+                 (close [_] (conn/stop! conn))
+                 p/PedestalConnector
+                 (start-connector! [_] (p/start-connector! conn))
+                 (stop-connector! [_] (p/stop-connector! conn))
+                 (test-request [_ ring-request] (p/test-request conn ring-request)))
+      assoc :connector-map connector-map :connector-opts connector-opts)))
+
+(defn send
```

Comment:
@hlship, I'd like to have your opinion on this `create-and-start!` and `send` functions.

I think they could be in a common testing namespace 
This set of functions allows
- Easily test without try/catch
- Using `:port 0` during the tests to avoid conflicts
- Switch between `create-connector` functions: even if we don't use it to test the same function on multiple backends, it allows the developer to easily compare the results between the backends.